### PR TITLE
Optimize enum class and change some java practice

### PR DESCRIPTION
**Pull request by yuqi1129**
## Description

This PR contains no fix and feature, is about java code optimization and contains the following contents:

1. Some java practice, for example , remove useless synchronized, remove duplication code and simplify code logic, see below

### CODE REVIEW

**Comment by HTHou** [comment]
Hi, actually we are working on refactoring the TsFile module and it's almost done. Can you please help us submit a PR to the `NewTsFile` branch? I think it may save a lot of works... :)

File: server/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java [hunk 1]
```diff
@@ -174,10 +175,11 @@ public void write(InsertTabletPlan insertTabletPlan, int start, int end) {
 
   @Override
   public boolean checkIfChunkDoesNotExist(String deviceId, String measurement) {
-    if (!memTableMap.containsKey(deviceId)) {
+    Map<String, IWritableMemChunk> memSeries = memTableMap.get(deviceId);
+    if (null == memSeries) {
       return true;
     }
-    Map<String, IWritableMemChunk> memSeries = memTableMap.get(deviceId);
+
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java:182]: else return false?

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/MetadataIndexNodeType.java [hunk 2]
```diff
@@ -33,7 +33,13 @@
  * LEAF_MEASUREMENT: leaf nodes of the index tree's device level, points to TimeseriesMetadata
  */
 public enum MetadataIndexNodeType {
-  INTERNAL_DEVICE, LEAF_DEVICE, INTERNAL_MEASUREMENT, LEAF_MEASUREMENT;
+  INTERNAL_DEVICE(0),LEAF_DEVICE(1), INTERNAL_MEASUREMENT(2), LEAF_MEASUREMENT(3);
+
+  private final int type;
```
Comment (jixuan1989) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/MetadataIndexNodeType.java:38]: use byte?

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/TSDataType.java [hunk 3]
```diff
@@ -24,7 +24,41 @@
...
+   * DOUBLE
+   */
+  DOUBLE(4),
+
+  /**
+   * TEXT
+   */
+  TEXT(5);
+
+  private final int type;
```
Comment (jixuan1989) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/TSDataType.java:57]: byte

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/TSEncoding.java [hunk 4]
```diff
@@ -20,7 +20,21 @@
...
+  DIFF(3),
+  TS_2DIFF(4),
+  BITMAP(5),
+  GORILLA_V1(6),
+  REGULAR(7),
+  GORILLA(8);
+
+  private final int type;
+
+  TSEncoding(int type) {
```
Comment (jixuan1989) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/TSEncoding.java:35]: byte

**Comment by jixuan1989** [comment]
any conflict with newTsfileBranch?

**Comment by jixuan1989** [comment]
> Kudos, SonarCloud Quality Gate passed!
> 
> [Bug](https://sonarcloud.io/project/issues?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=BUG)
> [Vulnerability](https://sonarcloud.io/project/issues?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=VULNERABILITY)
> [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=SECURITY_HOTSPOT)
> [Code Smell](https://sonarcloud.io/project/issues?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=CODE_SMELL) [6 Code Smells](https://sonarcloud.io/project/issues?id=apache_incubator-iotdb&pullRequest=2435&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=apache_incubator-iotdb&pullRequest=2435) No Coverage information
> [](https://sonarcloud.io/component_measures?id=apache_incubator-iotdb&pullRequest=2435&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=apache_incubator-iotdb&pullRequest=2435&metric=new_duplicated_lines_density&view=list)

code smell!

**Comment by HTHou** [comment]
> any conflict with newTsfileBranch? @HTHou

Almost the using byte you mentioned.

**Review by yuqi1129 — COMMENTED** [review]
@HTHou @qiaojialin 
Thank you for your feedback, i will compare it with #2184 and change accordingly.

**Comment by HTHou** [comment]
Hi, I checked the NewTsFile branch carefully. I found the if you change the `short` to `byte` in this PR, you may have to modify a lot of code in TsFile module. 

There are two ways, one is moving the TsFile part of this PR to another PR commit into `NewTsFile`. Another is just keeping the `short` here, once this PR merged into master, I'll resolve the conflicts in `NewTsFile`. 

I think either way is OK. Which one would you like?

**Comment by yuqi1129** [comment]
@HTHou Hi, I think the former is better, it will save us a lot of time and relatively simple.
To make sure, the `TsFile part of this PR` just means thel enum class changed in this  PR ? as other change in this PR is also about in the TsFile project.

File: server/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java (hunk 1 above)
Comment (jixuan1989) [reply server/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java:182]: my fault.  ignore the comment..

**Comment by HTHou** [comment]
> @HTHou Hi, I think the former is better, it will save us a lot of time and relatively simple.
> To make sure, the `TsFile part of this PR` just means thel enum class changed in this PR ? as other change in this PR is also about in the TsFile project.

What I mean is any thing about TsFile module. The change of memtable is another optimization I think :)

**Comment by yuqi1129** [comment]
> What I mean is anything about TsFile module. The change of memtable is another optimization I think :)
OK

**Comment by yuqi1129** [comment]
@HTHou , Changes about TsFile module have been in #2455 ,please take time review it if you are free, thanks

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**